### ISSUE TITLE
Image field null after upgrade

### ISSUE THREAD

**Issue by hiraldesai** [mikeedwards83/Glass.Mapper#177]
Hi guys,

Another minor issue with upgrade from v4.0.1.5 to v4.0.3.51. Can someone please confirm if this is expected or accidental breakage (this could very well be an expected behaviour and a developer oversight with null checking but wanted to double-check)?

All the fields with `Glass.Mapper.Sc.Fields.Image` as their type used to default to a non-null Image object with empty `Src` property on it. The new version doesn't seem to do that. I'm happy to add the null checks everywhere but how about existing `EditFrame` items in our views? Shouldn't they be working as before even when image is null?

This is the code that breaks with upgrade when there's no image configured in Sitecore for a given view rendering.
```
using (BeginEditFrame(Model, "Select Image", x => x.Image)
```
which I'm fixing by this
```
using (BeginEditFrame(Model, "Select Image", x => x.Image.HasValue() ? x.Image : new Glass.Mapper.Sc.Fields.Image()))
```

**Comment by cardinal252** [comment]
This one I am not sure on - @mikeedwards83 - any thoughts ?

**Comment by mikeedwards83** [comment]
Can you post your model?

I have replicated this and I don't get an issue.

**Comment by jules983** [comment]
Same issue here.. Any update?

**Comment by MiguelMinoldo** [comment]
Same issue here.. Any update? Thanks!

**Comment by MiguelMinoldo** [comment]
Just found this change on Glass.Mapper.Sc.DataMappers.SitecoreFieldImageMapper:

public override object GetField(Field field, SitecoreFieldConfiguration config, SitecoreDataMappingContext context)
    {
      if (Glass.Mapper.ExtensionMethods.IsNullOrEmpty(field.Value))
        return (object) null;
      Image img = new Image();
      ImageField field1 = new ImageField(field);
      SitecoreFieldImageMapper.MapToImage(img, field1);
      return (object) img;
    } 
	
public override object GetFieldValue(string fieldValue, SitecoreFieldConfiguration config, SitecoreDataMappingContext context)
    {
      Item innerItem = context.Service.Database.GetItem(new ID(fieldValue));
      if (innerItem == null)
        return (object) null;
      MediaItem imageItem = new MediaItem(innerItem);
      Image img = new Image();
      SitecoreFieldImageMapper.MapToImage(img, imageItem);
      return (object) img;
    } 

-> now, it's returining a Null object, causing some exceptions on my site, since we are only updating GlassMapper form V3 to V4.

Thanks in advance!

**Comment by mikeedwards83** [comment]
@MiguelMinoldo Is this problem related to the BeginEditFrame or to the RenderImage method?

**Comment by MiguelMinoldo** [comment]
In our case is with the render image as it's implemented on the current solution, that works fine with glass 3.

Kind regards

Mi Personal

---- Mike Edwards escribió ----

@MiguelMinoldo Is this problem related to the BeginEditFrame or to the RenderImage method?

---

**Comment by mikeedwards83** [comment]
Can you copy and paste the code into this issue.

**Comment by hiraldesai** [comment]
Sorry - this is in my list but haven't had a chance to get back to you - will do it soon.

**Comment by MiguelMinoldo** [comment]
Hi folks, here a view as example and the relevant code in the model:
@{
    var context = new Glass.Mapper.Sc.SitecoreContext();
    var page = context.GetCurrentItem<PageBase>();
    var children = context.GetItem<ContentFolder>(Model.Id).GetChildren<ContentItem>().Where(page.ShouldDisplayItem).ToList();
    var param = GetRenderingParameters<RenderingParams>();
}
	@foreach (var child in children)
	{
		<li>
			<div class="call-to-action-image">
				@using (BeginEditFrame("/sitecore/content/Applications/WebEdit/Child Edit Frame Buttons", child.Path))
				{
					using (BeginRenderLink(a => child.RelatedLink, null, true))
					{
						@RenderImage(a => child.Image, IsInEditingMode ? null : new { @height = child.Image.Height.ToString(), @width = child.Image.Width.ToString() }, true)
						<span class="label">@(IsInEditingMode ? Editable(a => child.RawHeading) : Html.Raw(child.Heading))</span>
					}
				}
			</div>
		</li>
	}

ContentItem Model:
.....
[SitecoreField(FieldId = "{7B59F8B4-791A-4C9B-8D9F-9144F19FE7B7}")]
public virtual Image Image { get; set; }
.....

Thanks in advance!

**Comment by mikeedwards83** [comment]
@MiguelMinoldo  There are mistakesin your lambda expression. Since around v3.3 we have started caching lambda expressions. Please see this page that explains it:

http://www.glass.lu/Mapper/Sc/Documentation/GlassHtml

**Comment by MiguelMinoldo** [comment]
Thanks Mike, I turned off the lambda caching but the issue still happens. Let me explain: since the migration to GlassMapper.Sc V4, the code: 
@RenderImage(a => child.Image, IsInEditingMode ? null : new { @height = child.Image.Height.ToString(), @width = child.Image.Width.ToString() }, true) 
is throwing an exception when the image field is empty in Sitecore, as I mentioned, I found that a change in "Glass.Mapper.Sc.DataMappers.SitecoreFieldImageMapper" is now returning a null object causing this kind of issues (i.e [NULL].Width).

Kind regards

**Comment by mikeedwards83** [comment]
@MiguelMinoldo Your best approach is to create a custom datamapper that inherits SitecoreFieldImageMapper and returns an Image class instead of null.

**Comment by MiguelMinoldo** [comment]
Thanks Mike, to avoid a big change on the solution updating every place where we are using the render image helper, I have fixed this issue by making this change on the model:

[SitecoreField(FieldId = "{XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX}")]
        public virtual Image SitecoreImage { get; set; }

        public virtual Image Image { get { return this.SitecoreImage ?? new Glass.Mapper.Sc.Fields.Image(); } }  

Thank you so much for your support.

Kind regards

**Comment by mikeedwards83** [comment]
I have added a flag to the config called ReturnEmptyImage. This allows you to control if an empty image object should be returned instead of null. This will be in the nightly build tomorrow and in the next full release.

**Comment by mikeedwards83** [comment]
Released in 4.0.4

**Comment by mikeedwards83** [comment]
See documentation:
http://www.glass.lu/Mapper/Sc/Documentation/EmptyImages
